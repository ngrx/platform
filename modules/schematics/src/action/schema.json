{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsNgRxAction",
  "title": "NgRx Action Options Schema",
  "type": "object",
  "properties": {
    "name": {
      "description": "The name of the action.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What should be the name of the action?"
    },
    "path": {
      "type": "string",
      "format": "path",
      "description": "The path to create the component.",
      "visible": false
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "aliases": ["p"]
    },
    "skipTests": {
      "type": "boolean",
      "description": "When true, does not create test files.",
      "default": false
    },
    "flat": {
      "type": "boolean",
      "default": true,
      "description": "Flag to indicate if a dir is created."
    },
    "group": {
      "type": "boolean",
      "default": false,
      "description": "Group actions file within 'actions' folder",
      "aliases": ["g"]
    },
    "api": {
      "type": "boolean",
      "default": false,
      "description": "Specifies if api success and failure actions should be generated.",
      "aliases": ["a"],
      "x-prompt": "Should we generate success and failure actions?"
    },
    "creators": {
      "type": "boolean",
      "default": true,
      "description": "Specifies whether to use creator functions for handling actions and reducers.",
      "aliases": ["c"],
      "x-prompt": "Do you want to use the create function?"
    }
  },
  "required": []
}
