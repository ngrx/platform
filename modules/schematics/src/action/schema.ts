export interface Schema {
  /**
   * The name of the component.
   */

  name: string;
  /**
   * The path to create the component.
   */

  path?: string;
  /**
   * The name of the project.
   */
  project?: string;
  /**
   * Specifies if a spec file is generated.
   */
  spec?: boolean;
  /**
   * Flag to indicate if a dir is created.
   */

  flat?: boolean;
  /**
   * Group actions file within 'actions' folder
   */

  group?: boolean;
}
