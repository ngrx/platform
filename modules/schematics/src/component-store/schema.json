{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsNgRxComponentStore",
  "title": "NgRx Component Store Options Schema",
  "type": "object",
  "properties": {
    "name": {
      "description": "The name of the component store.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What should be the name of the component store?"
    },
    "path": {
      "type": "string",
      "format": "path",
      "description": "The path to create the component store.",
      "visible": false
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "aliases": ["p"]
    },
    "flat": {
      "type": "boolean",
      "default": true,
      "description": "Flag to indicate if a dir is created."
    },
    "skipTests": {
      "type": "boolean",
      "default": false,
      "description": "When true, does not create test files."
    },
    "component": {
      "type": "string",
      "default": "",
      "description": "Allows specification of the declaring component.",
      "alias": "m",
      "subtype": "filepath",
      "x-prompt": "To which component (path) should the component store be provided in?"
    },
    "module": {
      "type": "string",
      "default": "",
      "description": "Allows specification of the declaring module.",
      "alias": "m",
      "subtype": "filepath",
      "x-prompt": "To which module (path) should the component store be provided in?"
    }
  },
  "required": []
}
