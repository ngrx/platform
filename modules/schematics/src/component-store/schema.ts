export interface Schema {
  /**
   * The name of the component store.
   */
  name: string;

  /**
   * The path to create the component store.
   */
  path?: string;

  /**
   * The name of the project.
   */
  project?: string;

  /**
   * Flag to indicate if a dir is created.
   */
  flat?: boolean;

  /**
   * When true, does not create test files.
   */
  skipTests?: boolean;

  /**
   * Allows specification of the declaring component.
   */
  component?: string;

  /**
   * Allows specification of the declaring module.
   */
  module?: string;
}
