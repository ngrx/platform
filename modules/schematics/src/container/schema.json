{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsNgRxContainer",
  "title": "NgRx Container Options Schema",
  "type": "object",
  "properties": {
    "path": {
      "type": "string",
      "format": "path",
      "description": "The path to create the component.",
      "visible": false
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "aliases": ["p"]
    },
    "name": {
      "type": "string",
      "description": "The name of the component.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What should be the name of the container component?"
    },
    "inlineStyle": {
      "description": "Specifies if the style will be in the ts file.",
      "type": "boolean",
      "alias": "s"
    },
    "inlineTemplate": {
      "description": "Specifies if the template will be in the ts file.",
      "type": "boolean",
      "alias": "t"
    },
    "viewEncapsulation": {
      "description": "Specifies the view encapsulation strategy.",
      "enum": ["Emulated", "Native", "None"],
      "type": "string",
      "alias": "v"
    },
    "changeDetection": {
      "description": "Specifies the change detection strategy.",
      "enum": ["Default", "OnPush"],
      "type": "string",
      "alias": "c"
    },
    "prefix": {
      "type": "string",
      "format": "html-selector",
      "description": "The prefix to apply to generated selectors.",
      "alias": "p"
    },
    "style": {
      "description": "The file extension or preprocessor to use for style files.",
      "type": "string"
    },
    "skipTests": {
      "type": "boolean",
      "description": "When true, does not create test files."
    },
    "flat": {
      "type": "boolean",
      "description": "Flag to indicate if a dir is created."
    },
    "skipImport": {
      "type": "boolean",
      "description": "Flag to skip the module import."
    },
    "selector": {
      "type": "string",
      "format": "html-selector",
      "description": "The selector to use for the component."
    },
    "module": {
      "type": "string",
      "description": "Allows specification of the declaring module.",
      "alias": "m"
    },
    "export": {
      "type": "boolean",
      "description": "Specifies if declaring module exports the component."
    },
    "state": {
      "type": "string",
      "description": "Specifies the path to the state exports."
    },
    "stateInterface": {
      "type": "string",
      "default": "State",
      "description": "Specifies the interface for the state."
    },
    "testDepth": {
      "description": "Specifies whether to create a unit test or an integration test.",
      "enum": ["unit", "integration"],
      "type": "string",
      "default": "integration"
    }
  },
  "required": []
}
