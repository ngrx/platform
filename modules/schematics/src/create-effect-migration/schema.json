{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsNgRxCreateEffectMigration",
  "title": "NgRx Effect Migration from @Effect to createEffect",
  "type": "object",
  "properties": {},
  "required": []
}
