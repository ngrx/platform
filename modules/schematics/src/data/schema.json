{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsNgRxData",
  "title": "NgRx Data Options Schema",
  "type": "object",
  "properties": {
    "name": {
      "description": "The name of the data entity.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      }
    },
    "path": {
      "type": "string",
      "format": "path",
      "description": "The path to create the service.",
      "visible": false
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "aliases": ["p"]
    },
    "skipTests": {
      "type": "boolean",
      "description": "When true, does not create test files.",
      "default": false
    },
    "flat": {
      "type": "boolean",
      "default": true,
      "description": "Flag to indicate if a dir is created."
    },
    "group": {
      "type": "boolean",
      "default": false,
      "description": "Group the services within relative subfolders",
      "aliases": ["g"]
    }
  },
  "required": []
}
