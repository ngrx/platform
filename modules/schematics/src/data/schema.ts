export interface Schema {
  /**
   * The name of the component.
   */
  name: string;

  /**
   * The path to create the component.
   */
  path?: string;

  /**
   * The name of the project.
   */
  project?: string;

  /**
   * When true, does not create test files.
   */
  skipTests?: boolean;

  /**
   * Flag to indicate if a dir is created.
   */

  flat?: boolean;

  /**
   * Group entity metadata files within 'data' folder
   */
  group?: boolean;
}
