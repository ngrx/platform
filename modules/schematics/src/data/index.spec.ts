import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import * as path from 'path';
import { Schema as DataOptions } from './schema';
import {
  getTestProjectPath,
  createWorkspace,
  defaultWorkspaceOptions,
  defaultAppOptions,
} from '@ngrx/schematics-core/testing';

describe('Data Schematic', () => {
  const schematicRunner = new SchematicTestRunner(
    '@ngrx/schematics',
    path.join(__dirname, '../../collection.json')
  );
  const defaultOptions: DataOptions = {
    name: 'foo',
    project: 'bar',
    group: false,
    flat: true,
  };

  const projectPath = getTestProjectPath();

  let appTree: UnitTestTree;

  beforeEach(async () => {
    appTree = await createWorkspace(schematicRunner, appTree);
  });

  it('should create the data service to a specified project if provided', async () => {
    const options = { ...defaultOptions, project: 'baz' };

    const specifiedProjectPath = getTestProjectPath(defaultWorkspaceOptions, {
      ...defaultAppOptions,
      name: 'baz',
    });

    const tree = await schematicRunner
      .runSchematicAsync('data', options, appTree)
      .toPromise();
    const files = tree.files;
    expect(
      files.indexOf(`${specifiedProjectPath}/src/lib/foo.service.ts`)
    ).toBeGreaterThanOrEqual(0);
    expect(
      files.indexOf(`${specifiedProjectPath}/src/lib/foo.ts`)
    ).toBeGreaterThanOrEqual(0);
  });

  it('should create the service and model files', async () => {
    const tree = await schematicRunner
      .runSchematicAsync('data', defaultOptions, appTree)
      .toPromise();
    expect(
      tree.files.indexOf(`${projectPath}/src/app/foo.service.ts`)
    ).toBeGreaterThanOrEqual(0);
    expect(
      tree.files.indexOf(`${projectPath}/src/app/foo.ts`)
    ).toBeGreaterThanOrEqual(0);
  });

  it('should create two files if skipTests is false(as it is by default)', async () => {
    const options = {
      ...defaultOptions,
    };
    const tree = await schematicRunner
      .runSchematicAsync('data', options, appTree)
      .toPromise();
    expect(
      tree.files.indexOf(`${projectPath}/src/app/foo.service.spec.ts`)
    ).toBeGreaterThanOrEqual(0);
    expect(
      tree.files.indexOf(`${projectPath}/src/app/foo.service.ts`)
    ).toBeGreaterThanOrEqual(0);
  });

  describe('service class', () => {
    it('should import the correct model', async () => {
      const options = { ...defaultOptions };
      const tree = await schematicRunner
        .runSchematicAsync('data', options, appTree)
        .toPromise();
      const fileContent = tree.readContent(
        `${projectPath}/src/app/foo.service.ts`
      );

      expect(fileContent).toMatch(/import { Foo } from '.\/foo';/);
    });

    it('should extending EntityCollectionServiceBase', async () => {
      const options = { ...defaultOptions };
      const tree = await schematicRunner
        .runSchematicAsync('data', options, appTree)
        .toPromise();
      const fileContent = tree.readContent(
        `${projectPath}/src/app/foo.service.ts`
      );

      expect(fileContent).toMatch(
        /export class FooService extends EntityCollectionServiceBase<Foo>/
      );

      expect(fileContent).toMatch(/super\('Foo', serviceElementsFactory\);/);
    });
  });

  it('should create model', async () => {
    const options = { ...defaultOptions };
    const tree = await schematicRunner
      .runSchematicAsync('data', options, appTree)
      .toPromise();
    const fileContent = tree.readContent(`${projectPath}/src/app/foo.ts`);

    expect(fileContent).toMatch(/export interface Foo {/);
    expect(fileContent).toMatch(/id\?: any;/);
  });

  it('should create spec class with right imports', async () => {
    const options = { ...defaultOptions };
    const tree = await schematicRunner
      .runSchematicAsync('data', options, appTree)
      .toPromise();
    const fileContent = tree.readContent(
      `${projectPath}/src/app/foo.service.spec.ts`
    );

    expect(fileContent).toMatch(/import { FooService } from '.\/foo.service'/);
  });
});
