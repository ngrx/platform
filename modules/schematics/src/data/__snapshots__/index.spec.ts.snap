// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Schematic should create a model interface 1`] = `
"export interface Foo {
  id?: unknown;
}
"
`;

exports[`Data Schematic should create a service class 1`] = `
"import { Injectable } from '@angular/core';
import {
  EntityCollectionServiceBase,
  EntityCollectionServiceElementsFactory
} from '@ngrx/data';
import { Foo } from './foo';

@Injectable({ providedIn: 'root' })
export class FooService extends EntityCollectionServiceBase<Foo> {
  constructor(serviceElementsFactory: EntityCollectionServiceElementsFactory) {
    super('Foo', serviceElementsFactory);
  }
}
"
`;

exports[`Data Schematic should create a spec class 1`] = `
"import { TestBed } from '@angular/core/testing';
import {
  EntityCollectionServiceElementsFactory
} from '@ngrx/data';
import { FooService } from './foo.service';

describe('FooService', () => {
  let service: FooService;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      providers: [
        EntityCollectionServiceElementsFactory,
        FooService
      ]
    });

    await TestBed.compileComponents();
  });

  beforeEach(() => {
    service = TestBed.inject(FooService);
  });

  it('should create an instance', () => {
    expect(service).toBeTruthy();
  });
});
"
`;
